### ISSUE TITLE
[encodeData] inconsistent return values

### ISSUE THREAD

**Issue by rryter** [ERC725Alliance/erc725.js#28]
As a developer I would expect the return values to be consistent with `encodeAllData`:

```javascript
const resultEncodeAllData = myERC725.encodeAllData({
  LSP3Profile: {
    hashFunction: 'keccak256(utf8)',
    hash: '0x820464ddfac1bec070cc14a8daf04129871d458f2ca94368aae8391311af6361',
    url: 'ifps://QmYr1VJLwerg6pEoscdhVGugo39pa6rycEZLjtRPDfW84UAx'
  }
});

const resultEncodeData = myERC725.encodeData('LSP3Profile', {
  hashFunction: 'keccak256(utf8)',
  hash: '0x820464ddfac1bec070cc14a8daf04129871d458f2ca94368aae8391311af6361',
  url: 'ifps://QmYr1VJLwerg6pEoscdhVGugo39pa6rycEZLjtRPDfW84UAx'
});

console.log(resultEncodeAllData[0]);
/* { 
  key: "0x5ef83ad9559033e6e941db7d7c495acdce616347d28e90c7ce47cbfcfcad3bc5", 
  value: "0x6f357c6a820464ddfac1bec070cc14a8daf04129871d458f2ca94368aae8391311af6361696670733a2f2f516d597231564a4c776572673670456f73636468564775676f3339706136727963455a4c6a7452504466573834554178"
} */

console.log(resultEncodeData);
// 0x6f357c6a820464ddfac1bec070cc14a8daf04129871d458f2ca94368aae8391311af6361696670733a2f2f516d597231564a4c776572673670456f73636468564775676f3339706136727963455a4c6a7452504466573834554178
```

`encodeData` should also return a key-value pair object.

## Demo
https://stackblitz.com/edit/erc725js-encode-data-issue?devtoolsheight=33&file=index.js

**Comment by robertdavid010** [comment]
i think it comes down to it whether it is more likely the developer would just want the value in a single case, since if they need the key they can just get it from the Schema, or hash the key name?  I can see how with `encodeData` that the developer would likely want to write this to the blockchain, in which case they would want the key as well. Also do we change `decodeData` results to match? It is much more likely in that case that the dev would not need the key. This would create inconsistency between the singular encode/decode/ method results, which could possibly confuse the developer? In practical terms this is probably a good thing to do, since it saves a line of code for the developer in return for learning to differentiate between the results of `encodeData` and `decodeData`. Would be good to check the pattern with other devs?

**Comment by robertdavid010** [comment]
Technically the API is consistent between the singular and the pluralized forms at the moment. Think we want to preserve consistency in the API, and then if this is a go-ahead, we should change `decodeData` to match and return a key/value pair as well, since then the developer will always know to use `resultVar.value` to get the value of the result for both singleton methods.

**Comment by rryter** [comment]
Will be resolved via #30
